### ISSUE TITLE
Resolving CUDA Out of Memory Error During Model Training with Full Gene Count on NVIDIA A100 GPU

### ISSUE THREAD

**Issue by tejuafonja** [Emad-COMBINE-lab/GRouNdGAN#6]
Hello,

I followed the instructions detailed on the [website](https://emad-combine-lab.github.io/GRouNdGAN/tutorial.html) but have been unable to successfully complete the training of the second stage of the model with the specified 1000 genes. I encounter a `torch.cuda.OutOfMemoryError: CUDA out of memory` error. Reducing the gene count to 10 allows the process to run smoothly. I am using an `NVIDIA A100-SXM4-40GB`, which I assumed would be sufficient since the tutorial states that the code can be run on a `Tesla V100-SXM2-32GB`.

Could you advise on adjustments I could make to run the code with the full number of genes as specified in the configuration file? Are there specific hyper-parameters that I should tweak to ensure the code runs to completion?

Thank you.

**Comment by YazdanZ** [comment]
Hi, 


We ran into the same problem as you when training on A100 GPUs. We mention the following in the manuscript - Methods:
"If the imposed GRN is relatively dense and contains more than 5000 edges, we set the depth to 2 and the width multiplier to 1 to be able to train on a single GPU. The details of hyperparameters and architectural choices are provided in Supplementary Table S2."


We verified that the model could run on `NVIDIA A100-SXM4-40GB` with the following change to the config file (`causal_gan.cfg`):
```
...
    depth per gene = 2
    width per gene = 1
...
```

Reducing the depth and the width multiplier of target generators will not have noticeable effects on data replication and GRN imposition.
